### ISSUE TITLE
String comparison in where clause not working properly

### ISSUE THREAD

**Issue by zhicwu** [ClickHouse/ClickHouse#8762]
This seems only related to JDBC/ODBC bridge, as `select * from (select 's' as s) where s = 's'` works but not `select * from jdbc('ch', '', 'select ''s'' as s') where s = 's'`. The same issue exists in both 19.11.8.46 and 19.16.10.44. A possible workaround is to explicitly convert the field to String first, for example: `toString(s) = 's'` or `concat(s, '') = 's'`.

To reproduce:
```bash
$ git clone https://github.com/zhicwu/clickhouse-jdbc-bridge-test.git
$ cd clickhouse-jdbc-bridge-test
$ docker-compose up -d
$ docker-compose exec ch-server clickhouse-client
ClickHouse client version 19.16.10.44 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 19.16.10 revision 54427.

ch-server :) select s, case when s = 's' then 1 else 0 end as compare from (select 's' as s) where s = 's';

SELECT 
    s, 
    multiIf(s = 's', 1, 0) AS compare
FROM 
(
    SELECT 's' AS s
)
WHERE s = 's'

┌─s─┬─compare─┐
│ s │       1 │
└───┴─────────┘

1 rows in set. Elapsed: 0.002 sec. 

ch-server :) select s, case when s = 's' then 1 else 0 end as compare from jdbc('ch', '', 'select ''s'' as s');

SELECT 
    s, 
    multiIf(s = 's', 1, 0) AS compare
FROM jdbc('ch', '', 'select \'s\' as s')

┌─s─┬─compare─┐
│ s │       1 │
└───┴─────────┘

1 rows in set. Elapsed: 0.035 sec. 

ch-server :) select s, case when s = 's' then 1 else 0 end as compare from jdbc('ch', '', 'select ''s'' as s') where s = 's';

SELECT 
    s, 
    multiIf(s = 's', 1, 0) AS compare
FROM jdbc('ch', '', 'select \'s\' as s')
WHERE s = 's'

Ok.

0 rows in set. Elapsed: 0.033 sec. 

ch-server :) select s, case when s = 's' then 1 else 0 end as compare from jdbc('ch', '', 'select ''s'' as s') where toString(s) = 's';

SELECT 
    s, 
    multiIf(s = 's', 1, 0) AS compare
FROM jdbc('ch', '', 'select \'s\' as s')
WHERE toString(s) = 's'

┌─s─┬─compare─┐
│ s │       1 │
└───┴─────────┘

1 rows in set. Elapsed: 0.036 sec. 

ch-server :) 
```

**Comment by alexey-milovidov** [comment]
@alex-krash no longer works on JDBC, will reassign.

**Comment by zhicwu** [comment]
Looks like the issue has been addressed as I didn't see that in `20.5.2.7`. Finally I can join multiple data sets via JDBC bridge in one query. Great work!
